src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -1159,6 +1159,16 @@ static object ConvertToBoolIfPossible(string arg)
                         showHelp: true);
                     return false;
                 }
+#if !UNIX
+                // Only do the .ps1 extension check on Windows since shebang is not supported
+                else if (!Path.GetExtension(_file).Equals(".ps1", StringComparison.OrdinalIgnoreCase))
```

Comment:
Really we have no need to do full path analysis for path extension: 
```suggestion
                else if (!_file.EndsWith(".ps1", StringComparison.OrdinalIgnoreCase))
```